### PR TITLE
API experiments

### PR DESCRIPTION
**Pull request by zmoon**
The goal here is to have one (or a few) modules that you can `use` to use the CATChem library, to provide an experience similar to when you use ESMF or netCDF in Fortran.

* `CC_` prefix except for the data types
* ...

### CODE REVIEW

**Comment by bbakernoaa** [comment]
this looks like it is moving along

File: src/api/catchem.F90 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+!> \file
+!! CATChem core data types and routines.
```
Comment (bbakernoaa) [inline src/api/catchem.F90:5]: should add a little more here for the grouping documentation 

`\defgroup API_CATChem`

File: src/api/catchem.F90 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   !
+   use precision_mod, only: cc_rk => fp
+   !
+   use species_mod, only: SpeciesType
+   !
+   use CCPr_Dust_Common_Mod, only: DustStateType
+   use CCPr_Dust_mod, only: cc_dust_init => CCPr_Dust_Init
+   use CCPr_Dust_mod, only: cc_dust_run => CCPr_Dust_Run
+   use CCPr_Dust_mod, only: cc_dust_finalize => CCPr_Dust_Finalize
+   !
```
Comment (bbakernoaa) [inline src/api/catchem.F90:39]: add sea salt 

```
    use CCPr_SeaSalt_Common_Mod, only: SeaSaltStateType
    use CCPr_SeaSalt_mod, only: cc_seasalt_init => CCPr_SeaSalt_Init
    use CCPr_SeaSalt_mod, only: cc_seasalt_run => CCPr_SeaSalt_Run
    use CCPr_SeaSalt_mod, only: cc_seasalt_finalize => CCPr_SeaSalt_Finalize
```

**Review by bbakernoaa — COMMENTED** [review]
Need to fix test_seasalt.f90 too

**Comment by zmoon** [comment]
@bbakernoaa I'm not sure how to get Doxygen to recognize and document the re-exported members.

**Comment by bbakernoaa** [comment]
I'll try to take a look and see if I can get it working

**Comment by bbakernoaa** [comment]
I say we go ahead and merge this and try to write the manual one later.  I think it will need some documentation for how to connect to a host model anyway.

**Review by bbakernoaa — APPROVED** [review]

**Comment by zmoon** [comment]
Sounds good.

**Pull request merged**